specification/cpim/resource-manager/Microsoft.AzureActiveDirectory/preview/2023-01-18-preview/externalIdentities.json

```diff
@@ -0,0 +1,1175 @@
+        },
+        "location": {
+          "type": "string",
+          "description": "The location in which the resource is hosted and data resides. Can be one of 'United States', 'Europe', 'Asia Pacific', or 'Australia'. Refer to [this documentation](https://aka.ms/B2CDataResidency) for more information.",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource Tags"
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData",
+          "readOnly": true
+        }
+      }
+    },
+    "B2CTenantResourceList": {
+      "description": "The collection of Azure AD B2C tenant resources",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "List of Azure AD B2C tenant resources",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/B2CTenantResource"
+          }
+        }
+      }
+    },
+    "CreateTenantRequestBody": {
+      "type": "object",
+      "description": "The information needed to create the Azure AD B2C tenant and corresponding Azure resource, which is used for billing purposes.",
+      "required": [
+        "location",
+        "sku",
+        "properties"
+      ],
+      "x-ms-azure-resource": true,
+      "properties": {
+        "location": {
+          "type": "string",
+          "description": "The location in which the resource is hosted and data resides. Can be one of 'United States', 'Europe', 'Asia Pacific', or 'Australia'. Refer to [this documentation](https://aka.ms/B2CDataResidency) for more information."
+        },
+        "properties": {
+          "type": "object",
+          "x-ms-client-flatten": true,
+          "properties": {
+            "createTenantProperties": {
+              "x-ms-client-flatten": true,
+              "$ref": "#/definitions/CreateTenantProperties"
+            }
+          }
+        },
+        "sku": {
+          "$ref": "#/definitions/B2CResourceSKU"
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource Tags"
+        }
+      }
+    },
+    "CreateTenantProperties": {
+      "type": "object",
+      "description": "These properties are used to create the Azure AD B2C tenant. These properties are not part of the Azure resource.",
+      "properties": {
+        "displayName": {
+          "description": "The display name of the Azure AD B2C tenant.",
+          "type": "string"
+        },
+        "countryCode": {
+          "$ref": "#/definitions/CountryCode"
+        },
+        "goLocal": {
```

Comment:
>goLocal [](http://example.com/codeflow?start=9&length=7)

It is recommended by ARM RPC to use enum instead of boolean, as enums are more future proof and extensible without needing to cause breaking changes.

See this: [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html) [Replace boolean/switch properties with better enum].

Reply:
Makes sense. I have updated the value to be an enum string and also only allow 1 choice.

Reply:
Actually we are going to rename the property to isGoLocalTenant and it will be a boolean. We do not expect this to ever change as GoLocal can only be opted in.

Reply:
@visingla-ms Can we move forward with this?